tests/RequestsTest.php

```diff
@@ -169,4 +172,19 @@ public function testTimeoutException() {
 		$this->expectExceptionMessage('timed out');
 		Requests::get(httpbin('/delay/3'), array(), $options);
 	}
+
+	public function testHasCapabilitiesSucceedsForDetectingSsl() {
+		$this->assertTrue(Requests::has_capabilities(array(Requests_Capability::SSL => true)));
+	}
+
+	public function testHasCapabilitiesFailsForUnsupportedCapabilities() {
+		$transports = new ReflectionProperty(Requests::class, 'transports');
+		$transports->setAccessible(true);
+		$transports->setValue(array(TestTransportMock::class));
```

Comment:
> Why are you using Reflection here instead of calling Requests::add_transport() ?

I'm using reflection because there is no way of replacing the existing transports, you can only add to them. And this specific test requires the capabilities call to fail on a certain condition, which cannot be done as soon as one of the already existing transports is registered.

> And on that note: should there be a Requests::remove_transport() method ?

Something seems to be missing for more control yes. Not sure removal is the best approach, though, maybe filtering in some form would be preferable.